test/Microsoft.ComponentDetection.Detectors.Tests/VcpkgComponentDetectorTests.cs

```diff
@@ -0,0 +1,174 @@
+            var components = detectedComponents.ToList();
+            var sbomComponent = (VcpkgComponent)components.FirstOrDefault()?.Component;
+
+            if (sbomComponent is null)
+            {
+                throw new AssertFailedException($"{nameof(sbomComponent)} is null");
+            }
+
+            Assert.AreEqual(1, components.Count());
+            Assert.AreEqual("nlohmann-json", sbomComponent.Name);
+            Assert.AreEqual("3.10.4", sbomComponent.Version);
+            Assert.AreEqual("0", sbomComponent.PortVersion);
+            Assert.AreEqual("SPDXRef-port", sbomComponent.SPDXID);
+            Assert.AreEqual("git+https://github.com/Microsoft/vcpkg#ports/nlohmann-json", sbomComponent.DownloadLocation);
+            Assert.AreEqual("pkg:vcpkg/nlohmann-json@3.10.4?port_version=0", sbomComponent.PackageUrl.ToString());
+        }
+
+        [TestMethod]
+        public async Task TestTinyxmlAndResource()
+        {
+            var spdxFile = @"{
+    ""SPDXID"": ""SPDXRef - DOCUMENT"",
+    ""documentNamespace"":
+        ""https://spdx.org/spdxdocs/tinyxml2-x64-linux-9.0.0-c99e4f03-5275-458b-8a69-b5f8dfa45f18"",
+    ""name"": ""tinyxml2:x64-linux@9.0.0 5c7679507def92c5c71df44aec08a90a5c749f7f805b3f0e8e70f5e8a5b1b8d0"",
+    ""packages"": [
+        {
+            ""name"": ""tinyxml2:x64-linux"",
+            ""SPDXID"": ""SPDXRef-binary"",
+            ""versionInfo"": ""5c7679507def92c5c71df44aec08a90a5c749f7f805b3f0e8e70f5e8a5b1b8d0"",
+            ""downloadLocation"": ""NONE"",
+            ""licenseConcluded"": ""NOASSERTION"",
+            ""licenseDeclared"": ""NOASSERTION"",
+            ""copyrightText"": ""NOASSERTION"",
+            ""comment"": ""This is a binary package built by vcpkg.""
+        },
+        {
+            ""SPDXID"": ""SPDXRef-resource-1"",
+            ""name"": ""leethomason/tinyxml2"",
+            ""downloadLocation"": ""git+https://github.com/leethomason/tinyxml2@9.0.0"",
+            ""licenseConcluded"": ""NOASSERTION"",
+            ""licenseDeclared"": ""NOASSERTION"",
+            ""copyrightText"": ""NOASSERTION"",
+            ""checksums"": [
+                {
+                    ""algorithm"": ""SHA512"",
+                    ""checksumValue"": ""9c5ce8131984690df302ca3e32314573b137180ed522c92fd631692979c942372a28f697fdb3d5e56bcf2d3dc596262b724d088153f3e1d721c9536f2a883367""
+                }
+            ]
+        }
+    ]
+}";
+            var (scanResult, componentRecorder) = await detectorTestUtility
+                                                  .WithFile("vcpkg.spdx.json", spdxFile)
+                                                    .ExecuteDetector();
+
+            Assert.AreEqual(ProcessingResultCode.Success, scanResult.ResultCode);
+
+            var detectedComponents = componentRecorder.GetDetectedComponents();
+            var components = detectedComponents.ToList();
+
+            Assert.AreEqual(2, components.Count());
+            {
```

Comment:
We don't need these curly brackets here. you will can just set the variable below without initializing it (without `var`)